MekHQ/src/mekhq/service/AutosaveService.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * AutosaveService.java
+ *
+ * Copyright (c) 2019 MekHQ Team. All rights reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.service;
+
+import megamek.common.logging.MMLogger;
+import mekhq.MekHQ;
+import mekhq.MekHqConstants;
+import mekhq.campaign.Campaign;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.prefs.Preferences;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+public class AutosaveService implements IAutosaveService {
+    private final Preferences userPreferences = Preferences.userRoot().node(MekHqConstants.AUTOSAVE_NODE);
+    private final MMLogger logger;
+
+    public AutosaveService(MMLogger logger) {
+        assert logger != null;
+
+        this.logger = logger;
+    }
+
+    @Override
+    public void requestDayAdvanceAutosave(Campaign campaign, int dayOfTheWeek) {
+        assert campaign != null;
+
+        if (this.isDailyAutosaveEnabled()) {
+            this.performAutosave(campaign);
+        } else if (dayOfTheWeek == Calendar.SUNDAY && this.isWeeklyAutosaveEnabled()) {
```

Comment:
Does MekHQ consider Sunday or Monday the first day of the week? I know AtB considers Monday to be the first day.

Reply:
Monday is the first day of the week I think, I wanted the autosave to happen just before the day changes to Monday (so the save happens on Sunday).

Reply:
Just tested, the save happens before the day changes. Do you think it would be better to save after the day changes? (so complete all the change day events/auto-stuff and then save?).

Reply:
I think before is best. This will help with debugging day start issues with repairs etc.